### ISSUE TITLE
tensorflow process killed without any error

### ISSUE THREAD

**Issue by LiweiPeng** [uber/horovod#609]
I am using horovod 1.5.0 + tensorflow for distributed training. Many times (about 1/10) I saw my training process was killed but there is no tensorflow or horovod error in the log. I also couldn't find any related error in OS logs. This happened in tf 1.8.0 and 1.11.0. 

Can this be related to horovod? If not, any insights? Thanks.

The following is the end of the log when this occurs: (The following is using 4 GPUs)

INFO:tensorflow:2018-10-30 10:26:03.272900: loss = 3.5446548, step = 5800 (145.448 sec)
INFO:tensorflow:2018-10-30 10:26:03.273053: loss = 3.5446548, step = 5800 (145.448 sec)
INFO:tensorflow:2018-10-30 10:26:03.273113: loss = 3.5446548, step = 5800 (145.448 sec)
INFO:tensorflow:2018-10-30 10:26:03.273203: loss = 3.5446548, step = 5800 (145.449 sec)
INFO:tensorflow:2018-10-30 10:28:28.958103: global_step/sec: 0.686408
INFO:tensorflow:2018-10-30 10:28:28.958218: global_step/sec: 0.686408
INFO:tensorflow:2018-10-30 10:28:28.958265: global_step/sec: 0.686408
INFO:tensorflow:2018-10-30 10:28:28.958430: global_step/sec: 0.686407
INFO:tensorflow:2018-10-30 10:28:28.958888: loss = 3.5169177, step = 5900 (145.686 sec)
INFO:tensorflow:2018-10-30 10:28:28.959016: loss = 3.5169177, step = 5900 (145.686 sec)
INFO:tensorflow:2018-10-30 10:28:28.959096: loss = 3.5169177, step = 5900 (145.686 sec)
INFO:tensorflow:2018-10-30 10:28:28.959279: loss = 3.5169177, step = 5900 (145.686 sec)

**Comment by alsrgv** [comment]
@LiweiPeng, could it be OOM killer or some monitoring process like that?

**Comment by LiweiPeng** [comment]
I searched all the logs while this occurred, there is no OOM related info. I am not aware of any monitoring process. And there is no core dump either. Very strange.

**Comment by LiweiPeng** [comment]
I did more testing and found this type of error occurs when OOM kills tensorflow. it is out of scope for Horovod. So close the issue.
